tests/test_get_circle_centers.py

```diff
@@ -7,29 +7,24 @@ def test_get_circle_centers():
     )
 
     assert circle_centers == [
```

Comment:
It's great that we finally added a test system, but this test is really bad.
It compares only against one of many valid results instead of verifying some properties of the result. Changes on the algorithm change the output and if the expected result has to be updated on every change the test is pretty useless.
Moreover, floating point errors can cause slightly different results, that's why the test is currently failing in the CI.

Some better tests would be
- Does each circle intersect the rectangle
- Is the rectangle fully covered (probably difficult)
- Circles don't overlap too much (threshold)
- Not too many circles for a given rectangle
- Are random points within the rectangle / search area also within at least one of the circles
- Check some edge cases